model/task/task.go

```diff
@@ -1380,12 +1380,18 @@ func (t *Task) SetGeneratedTasksToActivate(buildVariantName, taskName string) er
 func SetTasksScheduledTime(tasks []Task, scheduledTime time.Time) error {
 	ids := []string{}
 	for i := range tasks {
-		tasks[i].ScheduledTime = scheduledTime
-		ids = append(ids, tasks[i].Id)
+		// Skip tasks with scheduled time to prevent large updates
+		if tasks[i].ScheduledTime == utility.ZeroTime {
+			tasks[i].ScheduledTime = scheduledTime
+			ids = append(ids, tasks[i].Id)
+		}
 
 		// Display tasks are considered scheduled when their first exec task is scheduled
 		if tasks[i].IsPartOfDisplay() {
-			ids = append(ids, utility.FromStringPtr(tasks[i].DisplayTaskId))
+			displayTaskId := utility.FromStringPtr(tasks[i].DisplayTaskId)
+			if !utility.StringSliceContains(ids, displayTaskId) {
```

Comment:
I think this is a little strange, since we expect that display task IDs aren't being passed in above. Do you have reason to expect they are? If you just generally want to avoid dups to be careful, I think we could just call utility.UniqueStrings on ids before the query